xcode 26.4 compilation issues

Currently cannot compile EXLA due to compilation errors with Xcode 26.4

If I modify the Makefile with these flags, it works, but I imagine this would only be applicable for macOS, and don't know if this is the right thing to do.

```diff
# Build flags
# Note that XLA requires c++17, Fine as well
CFLAGS += -fPIC -I$(ERTS_INCLUDE_DIR) -I$(FINE_INCLUDE_DIR) -I$(XLA_INCLUDE_PATH) -Wall -Wno-sign-compare \
	 -Wno-unused-parameter -Wno-missing-field-initializers -Wno-comment \
-	 -std=c++17 -w
+	 -std=c++17 -w -Wno-invalid-specialization
```



```
$ mix compile
==> exla
Unpacking /Users/dbern/Library/Caches/xla/0.9.0/download/xla_extension-0.9.0-aarch64-darwin-cpu.tar.gz into foo
Using libexla.so from /Users/dbern/Library/Caches/xla/exla/elixir-1.19.5-erts-16.3-xla-0.9.0-exla-0.10.0-2sei57zdegbkw3jxycg5l2szgy/libexla.so
EXLA_CPU_ONLY is not set, checking for nvcc availability CUDA is not available.
c++ -fPIC -I/Users/dbern/.local/share/mise/installs/erlang/28.4/erts-16.3/include -Ifoo/deps/fine/incl
ude -Icache/xla_extension/include -Wall -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers -Wno-comment -std=c++17 -w -O3 -c c_src/exla/e
xla.cc -o cache/0.10.0/objs/exla.o
c++ -fPIC -I/Users/dbern/.local/share/mise/installs/erlang/28.4/erts-16.3/include -Ifoo/deps/fine/include -Icache/xla_extension/include -Wall -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers -Wno-comment -std=c++17 -w -O3 -c c_src/exla/e
xla_client.cc -o cache/0.10.0/objs/exla_client.o
c++ -fPIC -I/Users/dbern/.local/share/mise/installs/erlang/28.4/erts-16.3/include -Ifoo/deps/fine/include -Icache/xla_extension/include -Wall -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers -Wno-comment -std=c++17 -w -O3 -c c_src/exla/e
xla_mlir.cc -o cache/0.10.0/objs/exla_mlir.o
c++ -fPIC -I/Users/dbern/.local/share/mise/installs/erlang/28.4/erts-16.3/include -Ifoo/deps/fine/include -Icache/xla_extension/include -Wall -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers -Wno-comment -std=c++17 -w -O3 -c c_src/exla/i
pc.cc -o cache/0.10.0/objs/ipc.o
c++ -fPIC -I/Users/dbern/.local/share/mise/installs/erlang/28.4/erts-16.3/include -Ifoo/deps/fine/include -Icache/xla_extension/include -Wall -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers -Wno-comment -std=c++17 -w -O3 -c c_src/exla/c
ustom_calls/eigh_f64.cc -o cache/0.10.0/objs/custom_calls/eigh_f64.o
c++ -fPIC -I/Users/dbern/.local/share/mise/installs/erlang/28.4/erts-16.3/include -Ifoo/deps/fine/incl
ude -Icache/xla_extension/include -Wall -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers -Wno-comment -std=c++17 -w -O3 -c c_src/exla/custom_calls/eigh_f32.cc -o cache/0.10.0/objs/custom_calls/eigh_f32.o
c++ -fPIC -I/Users/dbern/.local/share/mise/installs/erlang/28.4/erts-16.3/include -Ifoo/deps/fine/incl
ude -Icache/xla_extension/include -Wall -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers -Wno-comment -std=c++17 -w -O3 -c c_src/exla/c
ustom_calls/lu_bf16.cc -o cache/0.10.0/objs/custom_calls/lu_bf16.o
c++ -fPIC
-I/Users/dbern/.local/share/mise/installs/erlang/28.4/erts-16.3/include -Ifoo/deps/fine/include -Icache/xla_extension/include -Wall -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers -Wno-comment -std=c++17 -w -O3 -c c_src/exla/c
ustom_calls/lu_f16.cc -o cache/0.10.0/objs/custom_calls/lu_f16.o
c++ -fPIC
-I/Users/dbern/.local/share/mise/installs/erlang/28.4/erts-16.3/include -Ifoo/deps/fine/include -Icache/xla_extension/include -Wall -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers -Wno-comment -std=c++17 -w -O3 -c c_src/exla/c
ustom_calls/lu_f32.cc -o cache/0.10.0/objs/custom_calls/lu_f32.o
c++ -fPIC
-I/Users/dbern/.local/share/mise/installs/erlang/28.4/erts-16.3/include -Ifoo/deps/fine/include -Icache/xla_extension/include -Wall -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers -Wno-comment -std=c++17 -w -O3 -c c_src/exla/c
ustom_calls/lu_f64.cc -o cache/0.10.0/objs/custom_calls/lu_f64.o
c++ -fPIC
-I/Users/dbern/.local/share/mise/installs/erlang/28.4/erts-16.3/include -Ifoo/deps/fine/include -Icache/xla_extension/include -Wall -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers -Wno-comment -std=c++17 -w -O3 -c c_src/exla/c
ustom_calls/qr_bf16.cc -o cache/0.10.0/objs/custom_calls/qr_bf16.o
c++ -fPIC
-I/Users/dbern/.local/share/mise/installs/erlang/28.4/erts-16.3/include -Ifoo/deps/fine/include -Icache/xla_extension/include -Wall -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers -Wno-comment -std=c++17 -w -O3 -c c_src/exla/c
ustom_calls/qr_f16.cc -o cache/0.10.0/objs/custom_calls/qr_f16.o
c++ -fPIC
-I/Users/dbern/.local/share/mise/installs/erlang/28.4/erts-16.3/include -Ifoo/deps/fine/include -Icache/xla_extension/include -Wall -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers -Wno-comment -std=c++17 -w -O3 -c c_src/exla/c
ustom_calls/qr_f32.cc -o cache/0.10.0/objs/custom_calls/qr_f32.o
c++ -fPIC
-I/Users/dbern/.local/share/mise/installs/erlang/28.4/erts-16.3/include -Ifoo/deps/fine/include -Icache/xla_extension/include -Wall -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers -Wno-comment -std=c++17 -w -O3 -c c_src/exla/c
ustom_calls/qr_f64.cc -o cache/0.10.0/objs/custom_calls/qr_f64.o
c++ -fPIC
-I/Users/dbern/.local/share/mise/installs/erlang/28.4/erts-16.3/include -Ifoo/deps/fine/include -Icache/xla_extension/include -Wall -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers -Wno-comment -std=c++17 -w -O3 -c c_src/exla/e
xla_cuda.cc -o cache/0.10.0/objs/exla_cuda.o
In file included from c_src/exla/exla.cc:6:
In file included from c_src/exla/exla_client.h:13:
In file included from cache/xla_extension/include/xla/pjrt/gpu/gpu_helpers.h:31:
In file included from cache/xla_extension/include/xla/tsl/framework/bfc_allocator.h:34:
In file included from cache/xla_extension/include/xla/tsl/framework/allocator.h:27:
cache/xla_extension/include/xla/tsl/framework/type_traits.h:109:8: error: 'is_signed' cannot be specialized: Users are not allowed to specialize this standard
 library entity [-Winvalid-specialization]
  109 | struct is_signed<tsl::qint8> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__type_traits/is_signed.h:25:8: note: marked
 '_Clang::no_specializations' here
   25 | struct _LIBCPP_NO_SPECIALIZATIONS is_signed : _BoolConstant<__is_signed(_Tp)> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:1014:9: note: expanded from macro '
_LIBCPP_NO_SPECIALIZATIONS'
 1014 |       [[_Clang::__no_specializations__("Users are not allowed to specialize this standard library entity")]]
      |         ^
In file included from c_src/exla/exla.cc:6:
In file included from c_src/exla/exla_client.h:13:
In file included from cache/xla_extension/include/xla/pjrt/gpu/gpu_helpers.h:31:
In file included from cache/xla_extension/include/xla/tsl/framework/bfc_allocator.h:34:
In file included from cache/xla_extension/include/xla/tsl/framework/allocator.h:27:
cache/xla_extension/include/xla/tsl/framework/type_traits.h:111:8: error: 'is_signed' cannot be specialized: Users are not allowed to specialize this standard
 library entity [-Winvalid-specialization]
  111 | struct is_signed<tsl::quint8> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__type_traits/is_signed.h:25:8: note: marked
 '_Clang::no_specializations' here
   25 | struct _LIBCPP_NO_SPECIALIZATIONS is_signed : _BoolConstant<__is_signed(_Tp)> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:1014:9: note: expanded from macro '
_LIBCPP_NO_SPECIALIZATIONS'
 1014 |       [[_Clang::__no_specializations__("Users are not allowed to specialize this standard library entity")]]
      |         ^
In file included from c_src/exla/exla.cc:6:
In file included from c_src/exla/exla_client.h:13:
In file included from cache/xla_extension/include/xla/pjrt/gpu/gpu_helpers.h:31:
In file included from cache/xla_extension/include/xla/tsl/framework/bfc_allocator.h:34:
In file included from cache/xla_extension/include/xla/tsl/framework/allocator.h:27:
cache/xla_extension/include/xla/tsl/framework/type_traits.h:113:8: error: 'is_signed' cannot be specialized: Users are not allowed to specialize this standard
 library entity [-Winvalid-specialization]
  113 | struct is_signed<tsl::qint16> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__type_traits/is_signed.h:25:8: note: marked
 '_Clang::no_specializations' here
   25 | struct _LIBCPP_NO_SPECIALIZATIONS is_signed : _BoolConstant<__is_signed(_Tp)> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:1014:9: note: expanded from macro '
_LIBCPP_NO_SPECIALIZATIONS'
 1014 |       [[_Clang::__no_specializations__("Users are not allowed to specialize this standard library entity")]]
      |         ^
In file included from c_src/exla/exla.cc:6:
In file included from c_src/exla/exla_client.h:13:
In file included from cache/xla_extension/include/xla/pjrt/gpu/gpu_helpers.h:31:
In file included from cache/xla_extension/include/xla/tsl/framework/bfc_allocator.h:34:
In file included from cache/xla_extension/include/xla/tsl/framework/allocator.h:27:
cache/xla_extension/include/xla/tsl/framework/type_traits.h:115:8: error: 'is_signed' cannot be specialized: Users are not allowed to specialize this standard
 library entity [-Winvalid-specialization]
  115 | struct is_signed<tsl::quint16> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__type_traits/is_signed.h:25:8: note: marked
 '_Clang::no_specializations' here
   25 | struct _LIBCPP_NO_SPECIALIZATIONS is_signed : _BoolConstant<__is_signed(_Tp)> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:1014:9: note: expanded from macro '
_LIBCPP_NO_SPECIALIZATIONS'
 1014 |       [[_Clang::__no_specializations__("Users are not allowed to specialize this standard library entity")]]
      |         ^
In file included from c_src/exla/exla.cc:6:
In file included from c_src/exla/exla_client.h:13:
In file included from cache/xla_extension/include/xla/pjrt/gpu/gpu_helpers.h:31:
In file included from cache/xla_extension/include/xla/tsl/framework/bfc_allocator.h:34:
In file included from cache/xla_extension/include/xla/tsl/framework/allocator.h:27:
cache/xla_extension/include/xla/tsl/framework/type_traits.h:117:8: error: 'is_signed' cannot be specialized: Users are not allowed to specialize this standard
 library entity [-Winvalid-specialization]
  117 | struct is_signed<tsl::qint32> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__type_traits/is_signed.h:25:8: note: marked
 '_Clang::no_specializations' here
   25 | struct _LIBCPP_NO_SPECIALIZATIONS is_signed : _BoolConstant<__is_signed(_Tp)> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:1014:9: note: expanded from macro '
_LIBCPP_NO_SPECIALIZATIONS'
 1014 |       [[_Clang::__no_specializations__("Users are not allowed to specialize this standard library entity")]]
      |         ^
In file included from c_src/exla/exla_client.cc:1:
In file included from c_src/exla/exla_client.h:13:
In file included from cache/xla_extension/include/xla/pjrt/gpu/gpu_helpers.h:31:
In file included from cache/xla_extension/include/xla/tsl/framework/bfc_allocator.h:34:
In file included from cache/xla_extension/include/xla/tsl/framework/allocator.h:27:
cache/xla_extension/include/xla/tsl/framework/type_traits.h:109:8: error: 'is_signed' cannot be specialized: Users are not allowed to specialize this standard
 library entity [-Winvalid-specialization]
  109 | struct is_signed<tsl::qint8> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__type_traits/is_signed.h:25:8: note: marked
 '_Clang::no_specializations' here
   25 | struct _LIBCPP_NO_SPECIALIZATIONS is_signed : _BoolConstant<__is_signed(_Tp)> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:1014:9: note: expanded from macro '
_LIBCPP_NO_SPECIALIZATIONS'
 1014 |       [[_Clang::__no_specializations__("Users are not allowed to specialize this standard library entity")]]
      |         ^
In file included from c_src/exla/exla_client.cc:1:
In file included from c_src/exla/exla_client.h:13:
In file included from cache/xla_extension/include/xla/pjrt/gpu/gpu_helpers.h:31:
In file included from cache/xla_extension/include/xla/tsl/framework/bfc_allocator.h:34:
In file included from cache/xla_extension/include/xla/tsl/framework/allocator.h:27:
cache/xla_extension/include/xla/tsl/framework/type_traits.h:111:8: error: 'is_signed' cannot be specialized: Users are not allowed to specialize this standard
 library entity [-Winvalid-specialization]
  111 | struct is_signed<tsl::quint8> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__type_traits/is_signed.h:25:8: note: marked
 '_Clang::no_specializations' here
   25 | struct _LIBCPP_NO_SPECIALIZATIONS is_signed : _BoolConstant<__is_signed(_Tp)> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:1014:9: note: expanded from macro '
_LIBCPP_NO_SPECIALIZATIONS'
 1014 |       [[_Clang::__no_specializations__("Users are not allowed to specialize this standard library entity")]]
      |         ^
In file included from c_src/exla/exla_client.cc:1:
In file included from c_src/exla/exla_client.h:13:
In file included from cache/xla_extension/include/xla/pjrt/gpu/gpu_helpers.h:31:
In file included from cache/xla_extension/include/xla/tsl/framework/bfc_allocator.h:34:
In file included from cache/xla_extension/include/xla/tsl/framework/allocator.h:27:
cache/xla_extension/include/xla/tsl/framework/type_traits.h:113:8: error: 'is_signed' cannot be specialized: Users are not allowed to specialize this standard
 library entity [-Winvalid-specialization]
  113 | struct is_signed<tsl::qint16> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__type_traits/is_signed.h:25:8: note: marked
 '_Clang::no_specializations' here
   25 | struct _LIBCPP_NO_SPECIALIZATIONS is_signed : _BoolConstant<__is_signed(_Tp)> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:1014:9: note: expanded from macro '
_LIBCPP_NO_SPECIALIZATIONS'
 1014 |       [[_Clang::__no_specializations__("Users are not allowed to specialize this standard library entity")]]
      |         ^
In file included from c_src/exla/exla_client.cc:1:
In file included from c_src/exla/exla_client.h:13:
In file included from cache/xla_extension/include/xla/pjrt/gpu/gpu_helpers.h:31:
In file included from cache/xla_extension/include/xla/tsl/framework/bfc_allocator.h:34:
In file included from cache/xla_extension/include/xla/tsl/framework/allocator.h:27:
cache/xla_extension/include/xla/tsl/framework/type_traits.h:115:8: error: 'is_signed' cannot be specialized: Users are not allowed to specialize this standard
 library entity [-Winvalid-specialization]
  115 | struct is_signed<tsl::quint16> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__type_traits/is_signed.h:25:8: note: marked
 '_Clang::no_specializations' here
   25 | struct _LIBCPP_NO_SPECIALIZATIONS is_signed : _BoolConstant<__is_signed(_Tp)> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:1014:9: note: expanded from macro '
_LIBCPP_NO_SPECIALIZATIONS'
 1014 |       [[_Clang::__no_specializations__("Users are not allowed to specialize this standard library entity")]]
      |         ^
In file included from c_src/exla/exla_client.cc:1:
In file included from c_src/exla/exla_client.h:13:
In file included from cache/xla_extension/include/xla/pjrt/gpu/gpu_helpers.h:31:
In file included from cache/xla_extension/include/xla/tsl/framework/bfc_allocator.h:34:
In file included from cache/xla_extension/include/xla/tsl/framework/allocator.h:27:
cache/xla_extension/include/xla/tsl/framework/type_traits.h:117:8: error: 'is_signed' cannot be specialized: Users are not allowed to specialize this standard
 library entity [-Winvalid-specialization]
  117 | struct is_signed<tsl::qint32> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__type_traits/is_signed.h:25:8: note: marked
 '_Clang::no_specializations' here
   25 | struct _LIBCPP_NO_SPECIALIZATIONS is_signed : _BoolConstant<__is_signed(_Tp)> {};
      |        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:1014:9: note: expanded from macro '
_LIBCPP_NO_SPECIALIZATIONS'
 1014 |       [[_Clang::__no_specializations__("Users are not allowed to specialize this standard library entity")]]
      |         ^
5 errors generated.
make: *** [cache/0.10.0/objs/exla.o] Error 1
make: *** Waiting for unfinished jobs....
5 errors generated.
make: *** [cache/0.10.0/objs/exla_client.o] Error 1
could not compile dependency :exla, "mix compile" failed. Errors may have been logged above. You can recompile this dependency with "mix deps.compile exla --force", update it with "mix deps.update exla" or clean it with "mix deps.clean exla"
```

--- a/exla/Makefile
+++ b/exla/Makefile
@@ -71,6 +71,8 @@ else
 endif
 
 ifeq ($(UNAME_S), Darwin)
+	# Xcode 26.4+ errors on XLA headers specializing std::is_signed.
+	CFLAGS += -Wno-invalid-specialization
 	LDFLAGS += -flat_namespace -undefined dynamic_lookup -rpath @loader_path/xla_extension/lib
 else
 	# Use a relative RPATH, so at runtime libexla.so looks for libxla_extension.so

